tar.go

```diff
@@ -96,6 +110,14 @@ func tar(ctx context.Context, enc FormatEncoder, path string, info os.FileInfo)
 		}
 		var items []FormatGoodbyeItem
 		for _, s := range stats {
+			if dev != 0 {
+				// one-file-system is set, skip other filesystems
+				st, ok := s.Sys().(*syscall.Stat_t)
+				if ok && st.Dev != dev {
```

Comment:
If we get here and Sys() is not *syscall.Stat_t, shouldn't we `continue` as well? Since it sort of indicates a different filesystem of sorts which is different from the one in `dev`.

Reply:
You are right, it should be `!ok || st.Dev != dev` instead of `ok && st.Dev != dev`.